### ISSUE TITLE
Javascript routing error

### ISSUE THREAD

**Issue by Portugao** [zikula/core#3807]
| Q                 | A
| ----------------- | ---
| Zikula Version    | 1.5.x
| PHP Version       | 5.6

### Expected behavior
Working with no error.

### Actual behavior
If I try in the hook section to assign a module to another I get a javascript routing error:
```
router.js:9 Uncaught Error: The route "zikula_hook_hook_togglesubscribeareastatus" does not exist.
    at K.f.i (http://zik15/web/bundles/fosjsrouting/js/router.js:9:313)
    at K.f.m (http://zik15/web/bundles/fosjsrouting/js/router.js:10:32)
    at subscriberAreaToggle (http://zik15/web/bundles/zikulahook/js/hookui.js:160:26)
    at bindSubscriberAreaToProviderArea (http://zik15/web/bundles/zikulahook/js/hookui.js:126:9)
    at HTMLDivElement.drop (http://zik15/web/bundles/zikulahook/js/hookui.js:392:17)
    at $.(anonymous function).(anonymous function)._trigger (http://zik15/web/bootstrap-jqueryui/bootstrap-jqueryui.min.js:5:13831)
    at $.(anonymous function).(anonymous function)._drop (http://zik15/web/bootstrap-jqueryui/bootstrap-jqueryui.min.js:5:52836)
    at $.(anonymous function).(anonymous function)._drop (http://zik15/web/bootstrap-jqueryui/bootstrap-jqueryui.min.js:5:6732)
    at $.(anonymous function).(anonymous function).<anonymous> (http://zik15/web/bootstrap-jqueryui/bootstrap-jqueryui.min.js:5:55303)
    at Function.each (http://zik15/web/jquery/jquery.min.js:2:2882)
```

### Steps to reproduce
try to assign the scribite (6.0 dev) ui hook to the pages module.

**Comment by craigh** [comment]
delete the `web/js` folder and see if it works. probably will.

**Comment by Portugao** [comment]
Seems to work.

**Comment by Portugao** [comment]
So is this a bug? Is deleting the folder web/js a workaround or the correct fix?

**Comment by MisterFau** [comment]
+1

**Comment by craigh** [comment]
yes it is a bug.

refs #3804 & #3273

**Comment by Guite** [comment]
> the javascript routing file must be regenerated when a module is installed, etc.

This should work in theory because the `InstallerListener` in the routes module reacts on `            CoreEvents::MODULE_POSTINSTALL`, `CoreEvents::MODULE_UPGRADE` and `CoreEvents::MODULE_REMOVE`.
https://github.com/zikula/core/blob/1.5/src/system/RoutesModule/Listener/InstallerListener.php

The result of the `dumpJsRoutes()` should be checked for errors though. If the result is not an empty string it should be added to the flash bag to give feedback about what did not work (e.g. missing permissions).

**Comment by shefik** [comment]
Deleting the file works for me too.

**Comment by Guite** [comment]
Closed by 78d9e049cb6ed94b30c8b24453d27bafac9b876f

**Comment by paustian** [comment]
As of 2.0.1 this is broken again. Creating a category causes a internal server error
